### PR TITLE
Added tests for JSON array of arrays and array of objects

### PR DESCRIPTION
**Pull request by selfcatering**
It took me a while to figure out how to write proper descriptions for JSON nested arrays of arrays and arrays of objects. Current JsonResponse seems to cover that quite well, but documentation is lacking instructions on how to properly write descriptions. This PR covers both scenarios and comments explain all the mandatory description elements. Hope that helps someone in the future.

### CODE REVIEW

File: tests/ResponseLocation/JsonLocationTest.php [hunk 1]
```diff
@@ -362,4 +362,176 @@ public function testVisitsNullResponseProperties()
...
+
+        $this->assertEquals($expected, $result->toArray());
+    }
+
+    /**
+     * @group ResponseLocation
+     */
+    public function testVisitsNestedArrayOfObjects()
+    {
+        $json = json_decode('{"scalar":"foo","nested":[{"bar":123,"baz":false},{"bar":345,"baz":true},{"bar":678,"baz":true}]}');
```
Comment (Konafets) [inline tests/ResponseLocation/JsonLocationTest.php]: Can you just copy the code from line 371. Its more clear and consistent.

Comment (selfcatering) [inline tests/ResponseLocation/JsonLocationTest.php]: Can you explain a bit more. Code from line 371 defines array of arrays. Line 469 with json_decode generates array of objects. Descriptions for both are a bit different but important enough to keep them separate.

Comment (Konafets) [inline tests/ResponseLocation/JsonLocationTest.php]: Yes, nevermind.

**Comment by Konafets** [comment]
@selfcatering, thanks for your contribution!

**Pull request merged**